mlrun/projects/project.py

```diff
@@ -2333,13 +2333,52 @@ def pull(
             clone_zip(url, self.spec.context, self._secrets)
 
     def create_remote(self, url, name="origin", branch=None):
-        """create remote for the project git
+        """Create remote for the project git
+
+         This method creates a new remote repository associated with the project's Git repository.
+         If a remote with the specified name already exists, it will not be overwritten.
+
+         If you wish to update the URL of an existing remote, use the `set_remote` method instead.
 
         :param url:    remote git url
         :param name:   name for the remote (default is 'origin')
         :param branch: Git branch to use as source
         """
         self._ensure_git_repo()
+        self.set_remote(url, name=name, branch=branch, overwrite=False)
+
+    def set_remote(self, url, name="origin", branch=None, overwrite=True):
+        """Create or update a remote for the project git repository.
+
+        This method allows you to manage remote repositories associated with the project.
+        It checks if a remote with the specified name already exists.
+
+        If a remote with the same name does not exist, it will be created.
+        If a remote with the same name already exists,
+        the behavior depends on the value of the 'overwrite' flag:
+
+        - If 'overwrite' is set to True (default), the existing remote will be overwritten with the new URL.
+        - If 'overwrite' is set to False, a ConflictError will be raised, indicating that the remote already exists
+          and cannot be overwritten.
```

Comment:
kind of a duplicate of lines 2367-2370, I think we can remove this and keep the information in the params below.